admin/Openedx_Woocommerce_Plugin_Admin.php

```diff
@@ -105,85 +110,138 @@ public function enqueue_styles() {
 		 * all of the hooks are defined in that particular class.
-         *
-         * The Openedx_Woocommerce_Plugin_Loader will then create the relationship
-         * between the defined hooks and the functions defined in this
-         * class.
-         */
-		
+		 *
+		 * The Openedx_Woocommerce_Plugin_Loader will then create the relationship
+		 * between the defined hooks and the functions defined in this
+		 * class.
+		 */
 	}
 
-     /**
-     * Register Enrollment Request custom post type 
-     *
-     * @since    1.0.0
-     */
-    public function register_enrollment_custom_post_type() {
+	/**
+	 * Register Enrollment Request custom post type 
+	 *
+	 * @since    1.0.0
+	 */
+	public function register_enrollment_custom_post_type()
+	{
 
 		$this->openedx_enrollment->register_enrollment_custom_post_type();
-
 	}
 
 	/**
 	 * Render Enrollment Request info form
 	 *
 	 * @since    1.0.0
 	 */
-	public function render_enrollment_info_form($post) {
+	public function render_enrollment_info_form($post)
+	{
 
 		$this->openedx_enrollment_info_form = new Openedx_Woocommerce_Plugin_Enrollment_Info_Form($post);
+	}
+
+	/**
+	 * Wrapper function to register a new post type
+	 *
+	 * @param  string $post_type   Post type name.
+	 * @param  string $plural      Post type item plural name.
+	 * @param  string $single      Post type item single name.
+	 * @param  string $description Description of post type.
+	 * @return object              Post type class object
+	 */
+	public function register_post_type($post_type = '', $plural = '', $single = '', $description = '', $options = array())
+	{
+
+		if (!$post_type || !$plural || !$single) {
+			return;
+		}
+
+		$post_type = $this->createPostType($post_type, $plural, $single, $description, $options);
+
+		return $post_type;
+	}
+
+	/**
+	 * Create a new instance of the Openedx_Woocommerce_Plugin_Post_Type class and register a new post type.
+	 *
+	 * @param  string $post_type   Post type name.
+	 * @param  string $plural      Post type item plural name.
+	 * @param  string $single      Post type item single name.
+	 * @param  string $description Description of the post type.
+	 * @param  array  $options     Additional options for the post type.
+	 * @return object              Post type class object.
+	 */
+	public function createPostType($post_type = '', $plural = '', $single = '', $description = '', $options = array())
+	{
+
+		return new Openedx_Woocommerce_Plugin_Post_Type($post_type, $plural, $single, $description, $options);
+	}
 
+	/**
+	 * Register course ID and mode fields for product
+	 *
+	 * @since    1.1.1
+	 */
+	function add_custom_product_fields()
+	{
+		global $post;
+
+		echo '<div class="options_group">';
+
+		echo '<p class="form-field">' . __('Only use these fields if the product is an Open edX course.', 'woocommerce') . '</p>';
+
+		woocommerce_wp_text_input(array(
+			'id'          => '_course_id',
+			'label'       => __('Course ID', 'woocommerce'),
```

Comment:
```suggestion
			'label'       => __('Open edX Course ID', 'woocommerce'),
```
It is a recommendation to be more specific.